### PR TITLE
[13.0] [FIX] stock_picking_product_barcode_report: Avoid empty data on wizard

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT27703

With next change we avoid the problem shown in the picture.

image

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I would add a `<p>` in the view saying that here only appears products with barcode defined.

**Comment by CarlosRoca13** [comment]
Done @pedrobaeza

File: stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing_view.xml [hunk 1]
```diff
@@ -11,6 +11,7 @@
                     <strong>Barcode Format</strong>
                 </span>
                 <field name="barcode_format" />
+                <p>Here only appears products with barcode defined</p>
```
Comment (pedrobaeza) [inline stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing_view.xml]:
```suggestion
                <p>It only appears here products with barcode defined</p>
```

File: stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def _onchange_picking_ids(self):
         ].default_get(line_fields)
         for picking_id in self.picking_ids.ids:
             picking = self.env["stock.picking"].browse(picking_id)
-            for move_line in picking.move_line_ids:
+            for move_line in picking.move_line_ids.filtered("product_id.barcode"):
```
Comment (sergio-teruel) [inline stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing.py:63]: @pedrobaeza What is better for performance, filter records or a continue inside the bucle?

Comment (pedrobaeza) [reply stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing.py:63]: `continue`, but sometimes it doesn't worth the less readable code for performance.

**Review by pedrobaeza — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

File: stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing.py (hunk 2 above)
Comment (sergio-teruel) [reply stock_picking_product_barcode_report/wizard/stock_barcode_selection_printing.py:63]: Oks, thanks

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**